### ISSUE TITLE
Name attribute doesn't depend on HtmlFieldPrefix

### ISSUE THREAD

**Issue by MarienMonnier** [Bitmapped/MvcEnumFlags#1]
In this current implementation, the name attribute doesn't depend on the HtmlFieldPrefix like the id.
So when you create have a loop around, the name attribute is always the same, instead of being Prefix[0].Name, Prefix[1].Name, ....

The only change I made was : 

```
checkbox.Attributes["name"] = ti.GetFullHtmlFieldName(string.Empty); // metadata.PropertyName;
```

**Comment by mitchej7** [comment]
I had a similar issue but where the property was two levels down. i.e model.AnotherProperty.EnumProperty.
The "AnotherProperty" isn't being set so the model binding doesn't kick in.

Needed to add: 
`var expressionText = ExpressionHelper.GetExpressionText((LambdaExpression) expression);
            string fullHtmlFieldName = htmlHelper.ViewContext.ViewData.TemplateInfo.GetFullHtmlFieldName(expressionText);`

And then set the name property using:
`checkbox.Attributes["name"] = fullHtmlFieldName;`

**Comment by Bitmapped** [comment]
Can you submit a pull request for me to look at?

**Comment by mitchej7** [comment]
I don't have permission to push a branch up to your repo. Do I need to fork it?

**Comment by Bitmapped** [comment]
Yes. Please fork it and then you can submit a pull request from there.

**Comment by Bitmapped** [comment]
I believe these issues should be fixed with PR #2. I'll update the NuGet release shortly.
